### DISCUSSION TITLE
Axi idWidth mismatch when using Axi4Upsizer as Axi bus master

### DISCUSSION THREAD

**Discussion by linux1299** [SpinalHDL/SpinalHDL#1164] [Q&A]
Hi, when I use Axi4Upsizer as Axi bus master, I get this compile error:
`assertion failed: Expect toplevel/??? : Stream[Axi4Ar] idWidth=4 <= toplevel/[Axi4SharedArbiter]/io_readInputs_0_ar : Stream[Axi4Ar] idWidth=3`
there is another master and 3 slaves in Axi bus. 
I have tried to remove the Axi4Upsizer type master, then there is no compile error.

**Comment by Dolu1990** [comment]
the AXI arbiter used extends the AXI id signal. It is used to route things back.

output id width = (inputs id width max) + log2up(inputs count)

So you need your AXI slaves id width to be bigger i think.
